packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -0,0 +1,301 @@
+/**
+@license
+Copyright 2019 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import '@material/mwc-notched-outline';
+
+import {addHasRemoveClass, classMap, FormElement, html, property, query, TemplateResult, PropertyValues} from '@material/mwc-base/form-element.js';
+import {floatingLabel, FloatingLabel} from '@material/mwc-floating-label';
+import {lineRipple, LineRipple} from '@material/mwc-line-ripple';
+import {NotchedOutline} from '@material/mwc-notched-outline';
+import {MDCTextFieldAdapter, MDCTextFieldInputAdapter, MDCTextFieldLabelAdapter, MDCTextFieldLineRippleAdapter, MDCTextFieldOutlineAdapter, MDCTextFieldRootAdapter} from '@material/textfield/adapter.js';
+import MDCTextFieldFoundation from '@material/textfield/foundation.js';
+
+import {characterCounter, CharacterCounter} from './character-counter/mwc-character-counter-directive.js';
+
+const passiveEvents = ['touchstart', 'touchmove', 'scroll', 'mousewheel'];
+
+export type HTMLInputElementType =
```

Comment:
Renamed to `TypeKeyword` as the HTML spec of HTMLInputElement.type has a table where they call these keywords. Got rid of HTMLInputElement part of the name because it's an implementation detail and the user does not need to know it is being used.

Reply:
also, they know it's going to be imported from this element so they know it'll be related to `mwc-textfield`. e.g. the affordance of

```ts
import {TypeKeyword} from '@material/mwc-textfield';
// or
import * as MWCTextfield from '@material/mwc-textfield';

const myType: MWCTextfield.TypeKeyword = 'email';
html`
  <mwc-textfield type=${myType}></mwc-textfield>
`
```